log4tango/config/config.cmake

```diff
@@ -27,32 +27,10 @@ endmacro()
 LOG4TANGO_CHECK_FUNCTION_EXISTS(ftime         HAVE_FTIME)
 LOG4TANGO_CHECK_FUNCTION_EXISTS(gettimeofday  HAVE_GETTIMEOFDAY)
 
-
-#check threads
-check_library_exists(pthread pthread_attr_get_np "" HAVE_PTHREAD_ATTR_GET_NP)
-check_library_exists(pthread pthread_getattr_np "" HAVE_PTHREAD_GETATTR_NP)
-check_library_exists(pthread pthread_create "" HAVE_PTHREAD_CREATE)
-check_include_file("pthread.h;pthread_np.h" HAVE_PTHREAD_NP_H)
-
 if(CMAKE_THREAD_LIBS_INIT)
     set(LOG4TANGO_HAVE_THREADING "/**/")
 endif()
```

Comment:
https://cmake.org/cmake/help/latest/module/FindThreads.html
> The thread library to use. This may be empty if the thread functions are provided by the system libraries and no special flags are needed to use them.

This check is probably bad for portability. Maybe we should check for `Threads_FOUND` instead?

Also, is it possible to use Tango without threads? After a quick check of omniORB source code it seems that they require threads (posix or ms or vxworks) and do not have any fallback implementation (compilation errors out with "Implementation header file incomplete"). And since log4tango is not standalone library, I suggest to do the following:
* require Threads_FOUND to be set here
* remove LOG4TANGO_HAVE_THREADING, remove DummyThread.hh/.cpp
* move log4tango/threading/StandardThreads.hh to log4tango/StandardThreads.hh if you decide to keep that file (I'd also rename to e.g. 'ThreadingUtils.hh)
* remove entire log4tango/threading directory

Reply:
Fair point. I've done that. And also reworked the `int64_t` check.

Reply:
@mliszcz I've created https://github.com/tango-controls/cppTango/issues/793 I think we should tackle that one in a separate PR. I don't see an issue with backwards compatibility.